### PR TITLE
Fix macos CI job

### PR DESCRIPTION
**Pull request by sfod**
- Add macos-13 image to OSX job.
- Use Python virtual env to install packages. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -325,7 +325,7 @@ jobs:
           python builder.pyz build -p ${{ env.PACKAGE_NAME }} --cmake-extra=-DBUILD_SHARED_LIBS=ON
 
   osx:
-    runs-on: macos-latest
+    runs-on: macos-13
```
Comment (xiazhvera) [inline .github/workflows/ci.yml:328]: As Github has more OS version support, we can run on multiple OS versions using the matrix, like what we did for linux. 

```suggestion
    runs-on: ${{ matrix.runner }}
    strategy:
      matrix:
        runner:
          - macos-13
          - macos-latest
```

Comment (sfod) [reply .github/workflows/ci.yml:328]: `macos-latest` currently fails. In the future, yes, that's a good idea.

Comment (xiazhvera) [reply .github/workflows/ci.yml:328]: I saw a workaround there https://github.com/orgs/Homebrew/discussions/5159
It is kind of hack, while I think it should be fine as we just use it in CI tests.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -337,7 +342,7 @@ jobs:
           ./builder build -p ${{ env.PACKAGE_NAME }} --spec=downstream
       - name: Running samples in CI setup
         run: |
-          python3 -m pip install boto3
+          python3 -m pip install boto3 --break-system-packages
```
Comment (sbSteveK) [inline .github/workflows/ci.yml:345]: Do we know why "--break-system-packages" is required for latest MacOS version? Would it be better to instead create and use a venv? Slightly worried that forcing the installation could cause a harder to track down problem in the future depending on what boto3 is overriding and/or the conflicts this is ignoring.

**Review by sbSteveK — APPROVED** [review]
Remove macos-latest and  --break-system-packaegs and ship. Optionally add macos-14.

**Pull request merged**